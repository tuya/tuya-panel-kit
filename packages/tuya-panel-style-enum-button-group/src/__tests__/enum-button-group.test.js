/**
 * @jest-environment jsdom
 */

 import React from 'react';
 import renderer from 'react-test-renderer';
 import TuyaRNSvgs from 'tuya-panel-kit/lib/components/iconfont/svg/defaultSvg';
 import { AcrylicEnumButtonGroup } from '../index';
 import { mount } from 'enzyme';

const data = {
  label: '开关1',
  icon: TuyaRNSvgs.power,
  activeIcon: TuyaRNSvgs.plus,
  key: '1',
};

const dataSource1 = new Array(15).fill(0).map((_, idx) => {
  return {
    ...data,
    label: `开关${idx + 1}`,
    key: String(idx + 1),
    disable: idx === 0,
  };
});

describe('EnumButtonGroup', () => {
  it('Acrylic', () => {
    const wrap = renderer.create(
      <AcrylicEnumButtonGroup data={dataSource1} defaultActiveKeys={['1']} />
    ).toJSON();
    expect(wrap).toMatchSnapshot();
  });

  it('radio select', () => {
    let activeKeys = ['1'];
    const changeActiveKeys = (key, next, data) => activeKeys = next;
    const wrap = mount(
      <AcrylicEnumButtonGroup handActiveKeyChange={changeActiveKeys} data={dataSource1} activeKeys={['1']} />
    );
    const touchInstance = wrap.find('TouchableOpacity').at(1);
    touchInstance.props().onPress();
    expect(activeKeys).toEqual(['2']);
  })

  it('multi select', () => {
    let activeKeys = ['1'];
    const changeActiveKeys = (key, next, data) => activeKeys = next;
    const wrap = mount(
      <AcrylicEnumButtonGroup type="multi" handActiveKeyChange={changeActiveKeys} data={dataSource1} activeKeys={['1']} />
    );
    const touchInstance = wrap.find('TouchableOpacity').at(1);
    touchInstance.props().onPress();
    expect(activeKeys).toEqual(['1', '2']);
  })
});