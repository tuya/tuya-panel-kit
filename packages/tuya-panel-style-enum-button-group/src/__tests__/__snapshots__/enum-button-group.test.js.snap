// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnumButtonGroup Acrylic 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0)",
        "borderRadius": 40,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "width": undefined,
      },
      undefined,
    ]
  }
/>
`;
